samples/DirectX/D3D12/HelloConstBuffer12.cs

```diff
@@ -0,0 +1,807 @@
+                }
+
+                if (adapter != null)
+                {
+                    adapter->Release();
+                }
+
+                if (swapChain != null)
+                {
+                    swapChain->Release();
+                }
+            }
+        }
+
+        // Load the sample assets.
+        private void LoadAssets()
+        {
+            Guid iid;
+            ID3DBlob* signature = null;
+            ID3DBlob* error = null;
+            ID3DBlob* vertexShader = null;
+            ID3DBlob* pixelShader = null;
+
+            try
+            {
+                // Create a root signature consisting of a descriptor table with a single CBV.
+                {
+                    D3D12_FEATURE_DATA_ROOT_SIGNATURE featureData = new D3D12_FEATURE_DATA_ROOT_SIGNATURE {
+
+                        // This is the highest version the sample supports. If CheckFeatureSupport succeeds, the HighestVersion returned will not be greater than this.
+                        HighestVersion = D3D_ROOT_SIGNATURE_VERSION_1_1
+                    };
+
+                    if (FAILED(_device->CheckFeatureSupport(D3D12_FEATURE.D3D12_FEATURE_ROOT_SIGNATURE, &featureData, (uint)sizeof(D3D12_FEATURE_DATA_ROOT_SIGNATURE))))
+                    {
+                        featureData.HighestVersion = D3D_ROOT_SIGNATURE_VERSION_1_0;
+                    }
+
+                    var ranges = new D3D12_DESCRIPTOR_RANGE1[1];
+                    const int RootParametersCount = 1;
+                    var rootParameters = new D3D12_ROOT_PARAMETER1[RootParametersCount];
+
+                    ranges[0].Init(D3D12_DESCRIPTOR_RANGE_TYPE.D3D12_DESCRIPTOR_RANGE_TYPE_CBV, 1, 0, 0, D3D12_DESCRIPTOR_RANGE_FLAGS.D3D12_DESCRIPTOR_RANGE_FLAG_DATA_STATIC);
+                    fixed (D3D12_DESCRIPTOR_RANGE1* pRanges = &ranges[0])
+                    {
+                        rootParameters[0].InitAsDescriptorTable(1, pRanges, D3D12_SHADER_VISIBILITY.D3D12_SHADER_VISIBILITY_VERTEX);
+                    }
+
+                    // Allow input layout and deny uneccessary access to certain pipeline stages.
+                    D3D12_ROOT_SIGNATURE_FLAGS rootSignatureFlags =
+                        D3D12_ROOT_SIGNATURE_FLAG_ALLOW_INPUT_ASSEMBLER_INPUT_LAYOUT |
+                        D3D12_ROOT_SIGNATURE_FLAG_DENY_HULL_SHADER_ROOT_ACCESS |
+                        D3D12_ROOT_SIGNATURE_FLAG_DENY_DOMAIN_SHADER_ROOT_ACCESS |
+                        D3D12_ROOT_SIGNATURE_FLAG_DENY_GEOMETRY_SHADER_ROOT_ACCESS |
+                        D3D12_ROOT_SIGNATURE_FLAG_DENY_PIXEL_SHADER_ROOT_ACCESS;
+
+                    var rootSignatureDesc = new D3D12_VERSIONED_ROOT_SIGNATURE_DESC();
+                    fixed (D3D12_ROOT_PARAMETER1* pRootParameters = &rootParameters[0])
+                    {
+                        rootSignatureDesc.Init_1_1(RootParametersCount, pRootParameters, 0, null, rootSignatureFlags);
+                    }
```

Comment:
Changing the above to a stackalloc will allow this to become
```suggestion
                        rootSignatureDesc.Init_1_1(RootParametersCount, rootParameters, 0, null, rootSignatureFlags);
```